center/src/main/java/com/microsoft/hydralab/center/service/TestDataService.java

```diff
@@ -245,6 +246,10 @@ public void checkTestDataAuthorization(SysUser requestor, String testId) {
 
     public List<PerformanceTestResultEntity> getPerformanceTestHistory(List<CriteriaType> queryParams) {
         Specification<PerformanceTestResultEntity> spec = new CriteriaTypeUtil<PerformanceTestResultEntity>().transferToSpecification(queryParams, false);
-        return performanceTestResultRepository.findAll(spec);
+        //TODO set page
+        PageRequest pageRequest = PageRequest.of(0, 10, Sort.by(Sort.Direction.DESC, "Date"));
+        List<PerformanceTestResultEntity> perfHistoryList = new ArrayList<>(performanceTestResultRepository.findAll(spec, pageRequest).getContent());
+        Collections.reverse(perfHistoryList);
```

Comment:
Just curious, why do we need reverse here.

Reply:
Because the list is ordered by date desc. If we don't reverse the list, the chart will be like this:
![image](https://user-images.githubusercontent.com/30514480/229727698-8dece29b-bf1b-465f-9255-2dc405020cec.png)

Reply:
Is the descending sorting related to the following line?
`PageRequest pageRequest = PageRequest.of(0, 10, Sort.by(Sort.Direction.DESC, "Date"));`

What will happen if we change `Sort.Direction.DESC` to `Sort.Direction.ASC`?

Reply:
`Sort.Direction.DESC` is for querying the lasted history data. If we change it to `Sort.Direction.ASC`, we can will get the earliest 10 data